src/components/ScrollOffsetContextProvider.tsx

```diff
@@ -0,0 +1,114 @@
+import type {ParamListBase, RouteProp} from '@react-navigation/native';
+import React, {createContext, useCallback, useEffect, useMemo, useRef} from 'react';
+import {withOnyx} from 'react-native-onyx';
+import type {NavigationPartialRoute, State} from '@libs/Navigation/types';
+import NAVIGATORS from '@src/NAVIGATORS';
+import ONYXKEYS from '@src/ONYXKEYS';
+import SCREENS from '@src/SCREENS';
+import type {PriorityMode} from '@src/types/onyx';
+
+type ScrollOffsetContextValue = {
+    /** Save scroll offset of flashlist on given screen */
+    saveScrollOffset: (route: RouteProp<ParamListBase>, scrollOffset: number) => void;
+
+    /** Get scroll offset value for given screen */
+    getScrollOffset: (route: RouteProp<ParamListBase>) => number | undefined;
+
+    /** Clean scroll offsets of screen that aren't anymore in the state */
+    cleanStaleScrollOffsets: (state: State) => void;
+};
+
+type ScrollOffsetContextProviderOnyxProps = {
+    /** The chat priority mode */
+    priorityMode: PriorityMode;
+};
+
+type ScrollOffsetContextProviderProps = ScrollOffsetContextProviderOnyxProps & {
+    /** Actual content wrapped by this component */
+    children: React.ReactNode;
+};
+
+const defaultValue: ScrollOffsetContextValue = {
+    saveScrollOffset: () => {},
+    getScrollOffset: () => undefined,
+    cleanStaleScrollOffsets: () => {},
+};
+
+const ScrollOffsetContext = createContext<ScrollOffsetContextValue>(defaultValue);
+
+/** This function is prepared to work with HOME screens. May need modification if we want to handle other types of screens. */
+function getKey(route: RouteProp<ParamListBase> | NavigationPartialRoute): string {
+    if (route.params && 'policyID' in route.params && typeof route.params.policyID === 'string') {
+        return `${route.name}-${route.params.policyID}`;
+    }
+    return `${route.name}-global`;
+}
+
+function ScrollOffsetContextProvider({children, priorityMode}: ScrollOffsetContextProviderProps) {
+    const scrollOffsetsRef = useRef<Record<string, number>>({});
+    const previousPriorityMode = useRef<PriorityMode>(priorityMode);
+
+    useEffect(() => {
+        if (previousPriorityMode.current === priorityMode) {
```

Comment:
We can use `usePrevious` hook to compare with prev props value.

Btw, is this reasonable approach to reset scroll of HOME screen here? This is general context provider.
I thought of handling in `LHNOptionsList`.

Reply:
I feel like having this in one place rather than scattering it in different files makes it easier to understand. But I am open to hear different opinions.